src/ng/directive/input.js

```diff
@@ -1563,6 +1565,49 @@ function rangeInputType(scope, element, attr, ctrl, $sniffer, $browser) {
     } :
     originalRender;
 
+  if (minAttrType) {
+    ctrl.$validators.min = minAttrType === 'min' && supportsRange ?
+      function noopMinValidator() {
+        // Since all browsers set the input to a valid value, we don't need to check validity
+        return true;
+      } :
+      // ngMin doesn't set the min attr, so the browser doesn't adjust the input value as setting min would
+      function minValidator(modelValue, viewValue) {
+        return ctrl.$isEmpty(viewValue) || isUndefined(minVal) || viewValue >= minVal;
+      };
+
+    setInitialValueAndObserver(minAttrType, 'min', minChange);
+  }
+
+  if (maxAttrType) {
+    ctrl.$validators.max = maxAttrType === 'max' && supportsRange ?
+      function noopMaxValidator() {
+        // Since all browsers set the input to a valid value, we don't need to check validity
+        return true;
+      } :
+      // ngMax doesn't set the max attr, so the browser doesn't adjust the input value as setting max would
+      function maxValidator(modelValue, viewValue) {
+        return ctrl.$isEmpty(viewValue) || isUndefined(maxVal) || viewValue <= maxVal;
+      };
+
+    setInitialValueAndObserver(maxAttrType, 'max', maxChange);
+  }
+
+  function setInitialValueAndObserver(attrType, attrName, changeFn) {
+    var initialAttrValue;
+    if (attrType === attrName) {
+      initialAttrValue = attr[attrName];
+      // Set the actual element attribute so that the browser can adjust the value based on
+      // the max value, which might be interpolated
+      element.attr(attrName, initialAttrValue);
+    }
+
+    // This initalizes the min/max value, so that non-support browsers validate with the correct
+    // values during the initial $render
+    changeFn(initialAttrValue);
```

Comment:
`initialAttrValue` will be undefined for `ngMin`/`ngMax`. Is this intended?

It is not clear why this function would do different things for `ngXyz` vs `xyz` attributes. If this is intended, you could add a short comment explaining what is going on.